src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImpl.java

```diff
@@ -0,0 +1,789 @@
+import org.kuali.rice.krad.util.ObjectUtils;
+import org.springframework.transaction.annotation.Transactional;
+
+import edu.cornell.kfs.pdp.CUPdpConstants;
+import edu.cornell.kfs.pdp.CUPdpParameterConstants;
+import edu.cornell.kfs.pdp.batch.PayeeACHAccountExtractStep;
+import edu.cornell.kfs.pdp.batch.service.PayeeACHAccountExtractService;
+import edu.cornell.kfs.pdp.businessobject.PayeeACHAccountExtractDetail;
+
+public class PayeeACHAccountExtractServiceImpl implements PayeeACHAccountExtractService {
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(PayeeACHAccountExtractServiceImpl.class);
+
+    private BatchInputFileService batchInputFileService;
+    private List<BatchInputFileType> batchInputFileTypes;
+    private ConfigurationService configurationService;
+    private ParameterService parameterService;
+    private MailService mailService;
+    private DocumentService documentService;
+    private DataDictionaryService dataDictionaryService;
+    private PersonService personService;
+    private SequenceAccessorService sequenceAccessorService;
+    private AchService achService;
+    private AchBankService achBankService;
+
+    private Map<String, List<String>> partialProcessingSummary;
+
+    // Portions of this method are based on code and logic from CustomerLoadServiceImpl.
+    @Transactional
+    @Override
+    public boolean processACHBatchDetails() {
+        LOG.info("processACHBatchDetails: Beginning processing of ACH input files");
+        
+        int numSuccess = 0;
+        int numPartial = 0;
+        int numFail = 0;
+        partialProcessingSummary = new HashMap<String, List<String>>();
+        
+        Map<String,BatchInputFileType> fileNamesToLoad = getListOfFilesToProcess();
+        LOG.info("processACHBatchDetails: Found " + fileNamesToLoad.size() + " file(s) to process.");
+        
+        List<String> processedFiles = new ArrayList<String>();
+        for (String inputFileName : fileNamesToLoad.keySet()) {
+            
+            LOG.info("processACHBatchDetails: Beginning processing of filename: " + inputFileName);
+            processedFiles.add(inputFileName);
+            
+            try {
+                List<String> errorList = new ArrayList();
+                errorList = loadACHBatchDetailFile(inputFileName, fileNamesToLoad.get(inputFileName));
+                if (errorList.isEmpty()) {
+                    LOG.info("processACHBatchDetails: Successfully loaded ACH input file");
+                    numSuccess++;
+                } else {
+                    LOG.warn("processACHBatchDetails: ACH input file contained "+ errorList.size() + " rows that could not be processed.");
+                    partialProcessingSummary.put(inputFileName, errorList);
+                    numPartial++;
+                }
+            } catch (Exception e) {
+                LOG.error("processACHBatchDetails: ", e);

```

Comment:
It would probably be good to have a little more in the message her to provide context. Previously, this said "Failed to load ACH input file" - something like that indicating what was happening when the exception was caught would be helpful I think.